index.js

```diff
@@ -143,8 +143,40 @@ class Dynamodb extends Datastore {
         });
     }
 
+    /**
+     * Scan records in the datastore
+     * @method scan
+     * @param  {Object}   config                Configuration object
+     * @param  {String}   config.table          Table name
+     * @param  {Object}   [config.params]       Record data
+     * @param  {Object}   config.paginate       Pagination parameters
+     * @param  {Number}   config.paginate.count Number of items per page
+     * @param  {Number}   config.paginate.page  Specific page of the set to return
+     * @param  {Function} callback              fn(err, data)
+     *                                          err - Error object
+     *                                          data - List of records in the table
+     */
     scan(config, callback) {
-        callback();
+        const client = this.client[config.table];
+        const endIndex = config.paginate.page * config.paginate.count;
+        const startIndex = (config.paginate.page - 1) * config.paginate.count;
+
+        if (!client) {
+            const err = new Error(`Invalid table name "${config.table}"`);
```

Comment:
We will keep the error message for now and work to make the two datastores consistent in another pull request